integration-tests/app_id_test.go

```diff
@@ -7,110 +7,116 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 )
 
-func TestAppIdSuccess(t *testing.T) {
+type AppCredentialTestSuite struct {
+	suite.Suite
+}
+
+func (suite *AppCredentialTestSuite) SetupSuite() {
 	err := StartContainers("./configs/app-id/docker-compose.yaml")
-	// Sleep required in github actions to prevent failing of tests in case of services getting late to start.
 	time.Sleep(10 * time.Second)
 	if err != nil {
 		fmt.Printf("Error: %v", err)
-		t.Fatalf("Error starting docker-compose: %v", err)
+		suite.Errorf(err, "Error starting docker-compose: %v")
 	}
+}
 
+func TestAppCredentialSuite(t *testing.T) {
+	fmt.Println("Running AppCredentialTestSuite")
+	suite.Run(t, new(AppCredentialTestSuite))
+}
+
+func (suite *AppCredentialTestSuite) TearDownSuite() {
+	fmt.Println("Stopping AppCredentialTestSuite")
+	errStop := StopContainers("./configs/app-id/docker-compose.yaml")
+	if errStop != nil {
+		fmt.Printf("Error stopping : %v", errStop)
+	}
+}
```

Comment:
Not important, but just a minor note for your own thoughts: when you have a method that does not require the self object (ie. `suite` in this case), that usually indicates that either the method should be a free standing function because it has no relation to the object or that the object is lacking some relationship to the method. In some cases there might arguably exist a point about namespacing/grouping together related functionality, but it's usually the case that there's a hidden object to be discovered there.

Reply:
yeah. `StartContainers` and `StopContainers` are helper methods that are shared by all the test suites. So didn't implement them as receiver functions.

Reply:
@unleashed Sorry I was saying something else in the comment. So `TearDownSuite` requires a pointer receiver because it's a method that is defined in the TestSuite interface. So it's something like you mentioned regarding name-spacing and grouping.